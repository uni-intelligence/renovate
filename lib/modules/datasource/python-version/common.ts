export const defaultRegistryUrl =
  'https://www.python.org/api/v2/downloads/release';

export const datasource = 'python-version';
