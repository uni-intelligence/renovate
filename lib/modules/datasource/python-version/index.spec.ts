import { getPkgReleases } from '..';
import { Fixtures } from '../../../../test/fixtures';
import * as httpMock from '../../../../test/http-mock';
import { EXTERNAL_HOST_ERROR } from '../../../constants/error-messages';
import { datasource, defaultRegistryUrl } from './common';

describe('modules/datasource/python-version/index', () => {
  describe('getReleases', () => {
    it('throws for 500', async () => {
      httpMock.scope(defaultRegistryUrl).get('/').reply(500);
      await expect(
        getPkgReleases({
          datasource,
          packageName: 'python',
        }),
      ).rejects.toThrow(EXTERNAL_HOST_ERROR);
    });

    it('returns null for error', async () => {
      httpMock.scope(defaultRegistryUrl).get('/').replyWithError('error');
      expect(
        await getPkgReleases({
          datasource,
          packageName: 'python',
        }),
      ).toBeNull();
    });

    it('returns null for empty 200 OK', async () => {
      httpMock.scope(defaultRegistryUrl).get('/').reply(200, []);
      expect(
        await getPkgReleases({
          datasource,
          packageName: 'python',
        }),
      ).toBeNull();
    });

    it('processes real data', async () => {
      httpMock
        .scope(defaultRegistryUrl)
        .get('/')
        .reply(200, Fixtures.get('release.json'));
      const res = await getPkgReleases({
        datasource,
        packageName: 'python',
      });
      expect(res).toMatchSnapshot();
      expect(res?.releases).toHaveLength(367);
    });
  });
});
