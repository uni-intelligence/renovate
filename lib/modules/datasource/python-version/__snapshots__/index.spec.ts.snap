// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`modules/datasource/python-version/index getReleases processes real data 1`] = `null`;
